### PR TITLE
WIP fix(coeditor): Allow configuring CKEditor via view extension

### LINKED ISSUE #7131: Pass ckeditor config array through a hook
Would be nice to alter bits of this without replacing the whole view.

### PR DESCRIPTION
**Pull request by mrclay**
Fixes #7131
- [ ] move to master if there's sufficient interest in this

### CODE REVIEW

File: mod/ckeditor/views/default/js/elgg/ckeditor.php [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
-	
+
 	var elggCKEditor = {
 
 		/**
+		 * CKEditor configuration
+		 */
+		config: (function () {
+
+			<?php elgg_view('js/elgg/ckeditor/config') ?>
```
Comment (ewinslow) [inline mod/ckeditor/views/default/js/elgg/ckeditor.php]: I'd strongly prefer not mixing php and js -- why can't we just do a `require('elgg/ckeditor/config')`?

**Comment by ewinslow** [comment]
Requirejs always requests files with a `.js` appended. That is why you need to name your files module.js.php if you need PHP (but really, you shouldn't need PHP...)

**Comment by ewinslow** [comment]
Ah ok, I see, the point is to extend the view.

eeeehhh.... don't like. :(

**Comment by jdalsem** [comment]
This is BC breaking as .php wins over .js.php, so previous overrides of the js file will fail... This should go to 2.x then



> eeeehhh.... don't like. :(

suggestions?

**Pull request closed without merging**